sdk/src/utils.ts

```diff
@@ -226,6 +228,23 @@ export const getDeploymentData = async (
   }
 }
 
+export const calculatePriceImpact = (
+  startingSqrtPrice: SqrtPrice,
+  endingSqrtPrice: SqrtPrice
+): Percentage => {
+  const startingPrice = startingSqrtPrice.v * startingSqrtPrice.v
+  const endingPrice = endingSqrtPrice.v * endingSqrtPrice.v
+
+  const nominator = BigInt(Math.abs(Number(startingPrice - endingPrice)))
+  const denominator = BigInt(Math.max(Number(startingPrice), Number(endingPrice)))
```

Comment:
What about domain and limit `number.max_safe_integer`? IMO there will be safety to use native `bigint`. So:
max: `a > b ? a : b`
abs: `x < 0n ? -x : x`